aten/src/ATen/native/mkldnn/xpu/detail/Conv.cpp

```diff
@@ -120,6 +120,12 @@ sycl::event convolution(
   }
 #endif
 
+  auto& ctx = at::globalContext();
+  bool allow_tf32 = ctx.allowTF32Onednn();
+  if (allow_tf32) {
+    pattr.set_fpmath_mode(dnnl::fpmath_mode::tf32);
+  }
+
```

Comment:
Pls. extract the logic as a utility function.

```C++
inline void apply_tf32_if_allowed(dnnl::primitive_attr& primitive_attr) {
  auto& ctx = at::globalContext();
  bool allow_tf32 = ctx.allowTF32Onednn();
  if (allow_tf32) {
    pattr.set_fpmath_mode(dnnl::fpmath_mode::tf32);
  }
}
```

Reply:
Thanks for suggestion, the utils have been extracted at https://github.com/pytorch/pytorch/pull/137570/files#diff-e2daddb2823a7fbde7e1c8944ac3d266c1270580b73e8d18ae333149540aa875R453